pkg/roachpb/metadata.go

```diff
@@ -946,7 +957,50 @@ func (h *GCHint) ForwardLatestRangeDeleteTimestamp(ts hlc.Timestamp) bool {
 	return false
 }
 
-// ResetLatestRangeDeleteTimestamp resets delete range timestamp.
-func (h *GCHint) ResetLatestRangeDeleteTimestamp() {
-	h.LatestRangeDeleteTimestamp = hlc.Timestamp{}
+// AddDeleteTimestamp adds the given deletion timestamp into the hint. Returns
+// true iff the hint was updated.
+func (h *GCHint) AddDeleteTimestamp(ts hlc.Timestamp) bool {
```

Comment:
Maybe something like `ScheduleGCAt` instead, to make it clear that the caller wants to run GC at some time, and the hint will figure out the best way to handle that.

`AddDeleteTimestamp` implies that the caller would only ever want to do this when it deletes something, but that isn't true.

Reply:
Done.